sw/device/tests/sensor_ctrl_alerts.c

```diff
@@ -182,6 +169,26 @@ bool test_main(void) {
                                              /*ecc_en*/ false,
                                              /*he_en*/ false);
 
+  // If all bits are set, then this must be the first iteration
+  uint32_t max = (1 << SENSOR_CTRL_PARAM_NUM_ALERT_EVENTS) - 1;
+  uint32_t new_val;
+  if (events_vector == -1) {
+    // Do not allow testing of all events at the same time
+    new_val = rand_testutils_gen32_range(1, max - 1);
```

Comment:
nit: Can you add small comment to describe why we are doing `max - 1` again here? It seems that the initial value of `max` already has a bound built into it.

Reply:
i updated this more to directly use a population count, this better matches my original intent.
Let me know if it makes sense.

Reply:
Makes sense to me. Thanks for updating.